tests/unittests/InterpreterTest.cpp

```diff
@@ -898,6 +898,89 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, bool isTrain) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  images.zero();
+  
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target, end;
+    target = nextRandInt(2, 5);
+    // to make sure testing set is not totally included in 
+    // training set
+    end = isTrain ? 6 : 7;
+
+    for (size_t x = 0; x < end; x++) {
```

Comment:
nit: I think I've already left this comment, that in fact we only need a single loop.
Target is known, target defines either row for line or row and column for cross.